src/libraries/System.Private.CoreLib/src/System/MemoryExtensions.cs

```diff
@@ -1220,6 +1222,30 @@ public static void Reverse<T>(this Span<T> span)
                 return;
             }
 
+            if (!RuntimeHelpers.IsReferenceOrContainsReferences<T>())
+            {
+                if (Unsafe.SizeOf<T>() == sizeof(byte))
+                {
+                    SpanHelpers.ReverseByteRef(ref Unsafe.As<T, byte>(ref MemoryMarshal.GetReference(span)), span.Length);
+                    return;
+                }
+                else if (Unsafe.SizeOf<T>() == sizeof(char))
+                {
+                    SpanHelpers.ReverseCharRef(ref Unsafe.As<T, char>(ref MemoryMarshal.GetReference(span)), span.Length);
+                    return;
+                }
+                else if (Unsafe.SizeOf<T>() == sizeof(int))
+                {
+                    SpanHelpers.ReverseInt32Ref(ref Unsafe.As<T, int>(ref MemoryMarshal.GetReference(span)), span.Length);
+                    return;
+                }
+                else if (Unsafe.SizeOf<T>() == sizeof(long))
+                {
+                    SpanHelpers.ReverseInt64Ref(ref Unsafe.As<T, long>(ref MemoryMarshal.GetReference(span)), span.Length);
+                    return;
+                }
+            }
+
```

Comment:
Does this `Reverse<T>(this Span<T> span)` method get inlined?  I'm assuming not.  Would it make sense to move the below fallback into its own method and then aggressively inline this wrapper, which will then end up being very small regardless of which path is taken?

Reply:
That makes sense. I haven't checked how it gets inlined, but I think you're right that it would be best to aggressively inline the wrapper and move the fallback case to a separate method.

I'll make the changes and push an update this afternoon.